tests/playbooks/tests_reapply.yml

```diff
@@ -46,7 +46,11 @@
                     - 192.0.2.72/31
                   dhcp4: false
                   auto6: false
-          ignore_errors: true
+            ignore_errors: true
+            ansible_managed_new: "{{ __network_new_header }}"
```

Comment:
These indentation changes look wrong.
Also, changing the test is surprising. I guess the change is intended for the tasks file.

Reply:
Right - `ansible_managed_new` should be passed as a key/value pair to the `network_connections` module, but `ignore_errors` should be outdented to the same level as `vars` e.g.
```yaml
          network_connections:
            ansible_managed_new: "{{ __network_new_header }}"
            ...
          ignore_errors: true
          vars:
            ...
```

Reply:
I believe that it is not correct to pass `ansible_managed_new` here at all since it is supposed to happen in `tasks/main.yml`.

Reply:
Thank you, @tyll and @richm. I'm fixing the indentation as @richm suggested.

> I believe that it is not correct to pass ansible_managed_new here at all since it is supposed to happen in tasks/main.yml.

Are you suggesting to remove `ansible_managed_new` at line 50 (now it is at line 49 before ignore_errors, though). Without `ansible_managed_new` here, tests_reapply_nm.yml fails as follows. I'd like to keep it to make the test pass if ok...
```
TASK [Show test_module_run] ***********************************************************************************
task path: /.../linux-system-roles/network/tests/playbooks/tests_reapply.yml:54
ok: [/home/nhosoi/.cache/linux-system-roles/rhel-8.qcow2] => {
    "test_module_run": {
        "changed": false,
        "failed": true,
        "msg": "missing required arguments: ansible_managed_new"
    }
}

TASK [Assert that reapply is found in log output] *************************************************************
task path: /.../linux-system-roles/network/tests/playbooks/tests_reapply.yml:57
fatal: [/.../.cache/linux-system-roles/rhel-8.qcow2]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'stderr'.
The error appears to be in '/.../linux-system-roles/network/tests/playbooks/tests_reapply.yml': line 57, column 11, but may
be elsewhere in the file depending on the exact syntax problem.
The offending line appears to be:
            var: test_module_run
        - name: Assert that reapply is found in log output
          ^ here
"}

TASK [network_connections] ************************************************************************************
task path: /.../linux-system-roles/network/tests/playbooks/tests_reapply.yml:64
fatal: [/.../.cache/linux-system-roles/rhel-8.qcow2]: FAILED! => {"changed": false, "msg": "missing required arguments: ansible_managed_new"}
...ignoring
```

Reply:
Ah, I see - the test `tests_reapply.yml` calls the module directly - it does not go through the role interface - so it must add the parameter - alternately, we could make the parameter optional, instead of required, and add some logic to the module code to skip ansible_managed handling - @tyll which would you prefer?

Reply:
Is it ok to leave this choice -- keeping `__ansible_managed` required and `__ansible_managed: "{{ __network_new_header }}"` in the test script?

Reply:
> Ah, I see - the test `tests_reapply.yml` calls the module directly - it does not go through the role interface - so it must add the parameter - alternately, we could make the parameter optional, instead of required, and add some logic to the module code to skip ansible_managed handling - @tyll which would you prefer?

Thank you for the clarification.

> Is it ok to leave this choice -- keeping `__ansible_managed` required and `__ansible_managed: "{{ __network_new_header }}"` in the test script?

it is correct to pass the parameter there, so they can be kept, just please with a different name (see my other comment).